lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,527 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "RuntimePlatformAgnosticPch.h"
+
+#ifdef INTL_ICU
+
+#include <stdint.h>
+// REVIEW (doilij): Where are these (non-'_t' types) typedef'd that is safe to include in PlatformAgnostic?
+typedef int16_t int16;
+typedef int32_t int32;
+typedef int64_t int64;
+typedef uint16_t uint16;
+typedef uint32_t uint32;
+typedef uint64_t uint64;
+
+#include "Codex/Utf8Codex.h"
+
+#ifndef _WIN32
+// REVIEW (doilij): The PCH allegedly defines enough stuff to get AssertMsg to work -- why was compile failing for this file?
+#ifdef AssertMsg
+#undef AssertMsg
+#endif
+#define AssertMsg(test, message)
+#ifdef AssertOrFailFastMsg
+#undef AssertOrFailFastMsg
+#endif
+#define AssertOrFailFastMsg(test, message)
+#endif // !_WIN32
+
+// REVIEW (doilij): copied from lib/Common/CommonPal.h because including that file broke the compile on Linux.
+// Only VC compiler support overflow guard
+#if defined(__GNUC__) || defined(__clang__)
+#define DECLSPEC_GUARD_OVERFLOW
+#else // Windows
+#define DECLSPEC_GUARD_OVERFLOW __declspec(guard(overflow))
+#endif
+
+#include "Common/MathUtil.h"
+#include "Core/AllocSizeMath.h"
+
+#include "Intl.h"
+#include "IPlatformAgnosticResource.h"
+
+#define U_STATIC_IMPLEMENTATION
+#define U_SHOW_CPLUSPLUS_API 1
+#pragma warning(push)
+#pragma warning(disable:4995) // deprecation warning
+#include <unicode/uloc.h>
+#include <unicode/numfmt.h>
+#include <unicode/enumset.h>
+#include <unicode/decimfmt.h>
+#pragma warning(pop)
+
+namespace PlatformAgnostic
+{
+namespace Intl
+{
+    using namespace PlatformAgnostic::Resource;
+
+    bool IsWellFormedLanguageTag(_In_z_ const char16 *languageTag, _In_ const charcount_t cch)
+    {
+        bool success = false;
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+        char icuLocaleId[ULOC_FULLNAME_CAPACITY] = { 0 };
+        char icuLangTag[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int32_t parsedLength = 0;
+        int32_t forLangTagResultLength = 0;
+        int32_t toLangTagResultLength = 0;
+        size_t inputLangTagUtf8SizeActual = 0;
+
+        // Allocate memory for the UTF8 output buffer. Need 3 bytes for each (code point + null) to satisfy SAL.
```

Comment:
Is there a reason you avoided the `WideToNarrow` class in `Utf8Helper.h` and did your own memory allocation / utf conversion calls?

Reply:
No particular reason. Wasn't aware, I guess. I'll look into it and we can update the implementation later.